File: spec/reverse_spec.rb
require 'rspec'
require_relative '../src/reverse'

describe "Given a string reverser" do

  describe "when given a string of at least 2 characters" do
    before (:each) do
      @subject = StringReverser.new('Simplicity is the glory of expression.')
    end

    it "has the original string" do
      expect(@subject.original_string).to eq('Simplicity is the glory of expression.')
    end
  end

  describe "when it is given a string of less than 2 characters" do

    it "returns an error message" do
      expect{StringReverser.new('a')}.to raise_error("The string must have 2 or more characters.")
    end
  end

end

File: spec/reverse_spec.js
(function(){
  'use strict'

  //= require ../src/reverse.js

  describe ("String reverser", function(){
    var subject;

    beforeEach(function(){
      var StringReverser = reverser.StringReverser;

      subject = new StringReverser('water');
    });

    it ("should be a thing", function(){
      expect(subject).toEqual(jasmine.anything());
    });

    it ("should pass back a string", function(){
      expect(subject.getString()).toEqual('water');
    });

    it ("should reverse the string", function(){
      expect (subject.reverseIt()).toEqual('retaw');
    });

  });
})();
